docs/lab_2.md

```diff
@@ -41,48 +55,60 @@ gcloud projects add-iam-policy-binding $PROJECT_ID \
 
 ## Create Cloud Storage bucket for Terraform state
 
-By default, Terraform stores state locally in a file named terraform.tfstate. This default configuration can make Terraform usage difficult for teams, especially when many users run Terraform at the same time and each machine has its own understanding of the current infrastructure. To help you avoid such issues, this section configures a remote state that points to a Cloud Storage bucket. Remote state is a feature of backends and, in this bootcamp, is configured in the backend.tf files.
+By default, Terraform stores state locally in a file named `terraform.tfstate`. This default configuration can make Terraform usage difficult for teams, especially when many users run Terraform at the same time and each machine has its own understanding of the current infrastructure. To help you avoid such issues, this section configures a remote state that points to a Cloud Storage bucket. Remote state is a feature of backends and, in this bootcamp, is configured in the `backend.tf` files.
 
-1. Start Cloud Shell
+1. Start Cloud Shell.
 
 > ![lab_2_cloud_shell](images/lab2-cloud-shell.jpg)
 
-2. In Cloud Shell, create the Cloud Storage bucket
+2. In Cloud Shell, create the Cloud Storage bucket using the following commands:
 
-```
+```bash
 PROJECT_ID=$(gcloud config get-value project)
 gsutil mb gs://${PROJECT_ID}-tfstate
 ```
 
-3. Enable Object Versioning to keep the history of your deployments
+3. Enable Object Versioning to keep the history of your deployments using the following command:
 
-```
+```bash
 gsutil versioning set on gs://${PROJECT_ID}-tfstate
 ```
 
-4. Replace the PROJECT_ID placeholder with the project ID in the backend.tf
+4. In GitHub, edit the `backend.tf` file and update the `bucket` to the Cloud Storage bucket that was just created (E.g., `sc-gc-devops-bootcamp-tfstate`).
+
+```json
+terraform {
+  backend "gcs" {
+    bucket = "sc-gc-devops-bootcamp-tfstate"
```

Comment:
Instruct the user to update the bucket to ${PROJECT_ID}-tfstate (otherwise bucket names wouldn't be unique)

Reply:
So the storage bucket names need to be unique, shouldn't we suggest to use something other than the `${PROJECT_ID}` or at least append a unique hash? I'd expect at least a few users will use the same project name that we used in the documentation/gif.

Reply:
@philbenjamin, I have updated the documentation to generate and add a random string (lowercase letters + numbers) to the bucket name. This should resolve any unique bucket creation name issues that may occur.

```bash
PROJECT_ID=$(gcloud config get-value project)
RAND=$(head /dev/urandom | tr -dc a-z0-9 | head -c 13)
gsutil mb gs://${PROJECT_ID}-${RAND}-tfstate
```

Reply:
Hey @kylelee24 , just saw this. So the project-id's in Google are globally unique. They are different from Project names which are not necessarily unique. So the random string addition is not necessary

Reply:
Hey @philbenjamin, my mistake. I thought the project id = project name. Looking at it, I see that the project id will be the same as the project name if the ID does not already exist. However if it exists, it appends a series of numbers after (E.g., `sc-gc-devops-bootcamp-284719`). I made the changes to ask the user to update the placeholder value of `PROJECT_ID` with the value of `${PROJECT_ID}`. I also updated the `backend.tf` file to have the `PROJECT_ID` placeholder.